metric-anomaly-data-model/src/main/avro/NotificationEvent.avdl

```diff
@@ -0,0 +1,15 @@
+@namespace("org.hypertrace.alert.engine.metric.anomaly.datamodel")
+protocol NotificationEventProtocol {
+  record NotificationEvent {
+    string tenant_id;
+    long event_time_millis;
+    map<string> action_event_metadata;
```

Comment:
nit : action_event_metadata -> metadata or notificaiton_event_metadata. But, too me it seems the name `metadata` suffice. Again if you would like to take it in follow up PR.